### PR TITLE
Dependent Fields:  POC with FormArray in parent expense form

### PR DESCRIPTION
**Pull request by yksolanki9**
New form group model for ref:
```
{
  dateOfSpend: "2023-01-16",
  ...,
  dependent_fields: [
    {
      label: 'Dependent field 0',
      data: {
        value: 'Open field 3',
        dependent_field_id: 3
      }
    },
    {
      label: 'Dependent field 3',
      data: {
        value: 'Open field 6',
        dependent_field_id: 6
      }
    },
    {
      label: 'Dependent field 6',
      data: {
        value: 'No further dependency',
        dependent_field_id: null
      }
    },
  ],
  ...
}
```

### CODE REVIEW

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 1]
```diff
@@ -2852,6 +2866,56 @@ export class AddEditExpensePage implements OnInit {
...
+    //     options: this.data.getFieldValuesById(0).data,
+    //     mandatory: this.data.expense_fields?.data[0].is_mandatory,
+    //     control: this.dependentFields.controls[0],
+    //   },
+    // ];
+
+    // this.fg.controls.dependent_fields.valueChanges.subscribe((val) => console.log('NEW VAL', val));
+  }
+
+  addDependentField(id: number) {
```
Comment (Julias0) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:2884]: id will always be formArray.length -1 right. Do we need this as a parameter?

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 2]
```diff
@@ -2852,6 +2866,56 @@ export class AddEditExpensePage implements OnInit {
...
+    console.log('ID IS', id, typeof id);
+    const formArray = this.fg.get('dependent_fields') as FormArray;
+
+    const newField = this.formBuilder.group({
+      fieldIndex: [formArray.length || 0],
```
Comment (Julias0) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:2889]: Why do you need this inside the form?

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 3]
```diff
@@ -2852,6 +2866,56 @@ export class AddEditExpensePage implements OnInit {
...
+    //     control: this.dependentFields.controls[0],
+    //   },
+    // ];
+
+    // this.fg.controls.dependent_fields.valueChanges.subscribe((val) => console.log('NEW VAL', val));
+  }
+
+  addDependentField(id: number) {
+    console.log('ID IS', id, typeof id);
+    const formArray = this.fg.get('dependent_fields') as FormArray;
```
Comment (Julias0) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:2886]: this.fg.controls. dependent_fields
This gives better intellisense support

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 4]
```diff
@@ -2852,6 +2866,56 @@ export class AddEditExpensePage implements OnInit {
...
+      mandatory: this.data.expense_fields?.data[id].is_mandatory,
+      control: newField,
+    });
+
+    formArray.push(newField, { emitEvent: false });
+  }
+
+  onDependentFieldChanged(data) {
+    console.log('FIELD CHANGES', data);
+    const updatedDependentField = (this.fg.get('dependent_fields') as FormArray).controls[data.fieldIndex] as FormGroup;
```
Comment (Julias0) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:2910]: You should define getters and methods for these

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts (hunk 1 above)
Comment (yksolanki9) [reply src/app/fyle/add-edit-expense/add-edit-expense.page.ts:2873]: Nai, id is the dependent field field based on the value the user selects. Have renamed it

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts (hunk 3 above)
Comment (yksolanki9) [reply src/app/fyle/add-edit-expense/add-edit-expense.page.ts:2874]: Fixed

**Review by Julias0 — CHANGES_REQUESTED** [review]
Sending back because POC

**Pull request closed without merging**